### PR TITLE
executor: fix IndexMergeReaderExecutor hang when to exit

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #41545

Problem Summary:

### What is changed and how it works?

It is found at https://github.com/pingcap/tidb/pull/41306.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/index_merge_reader.go [hunk 1]
```diff
@@ -360,12 +360,12 @@ func (e *IndexMergeReaderExecutor) startPartialIndexWorker(ctx context.Context,
 					SetPaging(e.paging).
 					SetFromInfoSchema(e.ctx.GetInfoSchema()).
 					SetClosestReplicaReadAdjuster(newClosestReadAdjuster(e.ctx, &builder.Request, e.partialNetDataSizes[workID]))
-
+			forLabel:
```
Comment (guo-shaoge) [inline executor/index_merge_reader.go:363]: Maybe open another issue.

**Review by XuHuaiyu — COMMENTED** [review]
Can we add a test for this fix

**Comment by guo-shaoge** [comment]
Will fix here https://github.com/pingcap/tidb/pull/41633

**Pull request closed without merging**